README.md

```diff
@@ -1,56 +1,71 @@
+
+_Note: For now the Cosmos-SDK only exists in [Golang](https://golang.org/), which means that developers can only develop SDK modules in Golang. In the future, we expect that the SDK will be implemented in other programming languages. The Cosmos team will also financially support their development._
+
+## Application architecture
+
+#### Modules
+
+A `module` is a fundamental unit in the Cosmos-SDK. Each module is an extension to the BaseApp functionalities that defines its own transactions, handles its own state as well as the state transition logic. The Cosmos-SDK has all the necessary pre-built modules to add functionality on top of a `BaseApp`, which is the template to build a blockchain dApp in Cosmos. Some of the most important ones are:
+
+- `Auth` - Defines a `BaseAccount` structure and how transaction signers are authenticated.
+- `Bank` - Defines how coins (i.e cryptocurrencies) are transferred.
+- `Governance` -  Governance related specifications including proposals and voting.
+- `Staking` - Proof of Stake related specifications including bonding and delegation transactions, inflation, fees, unbonding, etc.
+- `IBC` - Defines the intereoperability of blockchain zones according to the specifications of the [IBC Protocol](https://cosmos.network/whitepaper#inter-blockchain-communication-ibc).
+- Handlers for messages and transactions
+- REST and CLI for secure user interactions
+
+#### Directories
+
+The key directories of the SDK are:
+
+- `baseapp`: Defines the template for a basic application. It implements the [ABCI protocol](https://cosmos.network/whitepaper#abci) so that your Cosmos-SDK application can communicate with the underlying Tendermint node.
+- `client`: Command-Line to interface with the application.
+- `server`: Rest server to communicate with the node.
+- `examples`: Contains example on how to build a working application based on baseapp and module.
+- `store`: Contains code for the multistore (i.e state). Each module can create any number of KVStores from the multistore. Be careful to properly handle access rights to each store with appropriate keepers.
+- `types`: Common types required in any SDK-based application.
+- `x`(for e**X**tensions): Folder for storing the BaseApp module. You will find all the already-built modules in this directory. To use any of these modules, you just need to properly import them in your application.

```

Comment:
haha - always wondered about the x

Reply:
"properly import" is kinda iffy language (what is proper) - Also I'd remove use of the word `you` in this bullet